### DISCUSSION TITLE
Support for Redshift?

### DISCUSSION THREAD

**Discussion by jordanhamill** [amacneil/dbmate#361] [Feature Ideas]
Given that AWS Redshift is mostly compatible with Postgres I gave it a try but it failed immediately in `quotedMigrationsTableNameParts`.

> ERROR:  syntax error at or near "unnest"
> LINE 1: select unnest({public,test_schema});

Unfortunately Redshift doesn't support the array syntax and therefore the unnesting used  here:
```go
tableNameParts = append([]string{schema}, tableNameParts...)
	quotedNameParts, err := dbutil.QueryColumn(db, "select quote_ident(unnest($1::text[]))", pq.Array(tableNameParts))
```

Are there any alternatives that could replace this to achieve the same outcome?

**Comment by amacneil** [comment]
Interesting.

There is probably another way to achieve this. From memory that was the best way I could find to get a quoted array of identifiers. But I'm open to suggestions/PRs.

After fixing this (assuming nothing else is broken) it would really be best to set up an integration test in CI against redshift, to ensure it keeps working in the future.

**Comment by jordanhamill** [comment]
I appreciate this is not a great solution but I'm not sure I fully understand the usecase as well. 

I hacked around it by performing quoting on the client for my own use and everything else works. We don't use the schema dump. 
https://github.com/jordanhamill/dbmate/commit/1b1b476da5698cf3040470630b27ac551a0053e9#diff-d113ace4dc366910c6c6a5149162a9af2cb3056ed843238ae370835f77724ae5R378 (Golang is not my strong point)

After researching SQL only solutions, this is the best I could come up with which has some obvious limitations:
https://stackoverflow.com/questions/25112389/redshift-convert-comma-delimited-values-into-rows
```sql
with number_sequence as (
    select 1 as n
    union 
    select 2 as n
    union 
    select 3 as n
),
comma_separated as (
    select 'public,schema migrations' as value
),
unnested as (
    select quote_ident(split_part(cs.value, ',', n))
    from comma_separated cs
    cross join number_sequence
    where split_part(cs.value, ',', n) != ''
)
select * 
from unnested;
```

**Comment by carmoreno1** [comment]
At this moment is there some official solution to this case?

**Comment by amacneil** [comment]
I think if you use [v1.11.0](https://github.com/amacneil/dbmate/releases/tag/v1.11.0) this particular issue won't occur, since it seems related to the custom `--migrations-table` option introduced in v1.12.

I'm happy to accept a PR fixing this but don't currently have time to work on it myself.

**Comment by aterekhov-plr** [comment]
I suppose that isn't the best solution but it's very pity that we limited to use dbmate old version, so I prepare small PR  - https://github.com/amacneil/dbmate/pull/488
@amacneil I would appreciate any feedback, but it looks like we could continue support Redshift with this small improvement.

**Comment by dossy** [comment]
Duplicate of #198

**Comment by dossy** [comment]
Redshift support added in [v2.11.0](https://github.com/amacneil/dbmate/releases/tag/v2.11.0) via PR #488.
